File: 13-advanced/yegge/04-sumIntegers.js
var fs = require('fs');

var sumFile = function (filename) {
  // Returns the integer of string i or 0 if i is not a number
  var int = function (i) {
    i = parseInt(i);
    return isNaN(i) ? 0 : i;
  };

  var buffer = fs.readFileSync(filename);
  var lines = buffer.toString().split('\n');

  return lines.reduce(function (sum, i) {
    return sum + int(i);
  }, 0);
};

console.log( sumFile('numbers.txt') );

File: 13-advanced/yegge/04-sum_integers.rb
# Write a function that sums up integers from a text file, one int per line.
# ruby 04-sum_integers.rb numbers.txt

# sum = 0
# ARGF.each do |line|
#   sum += line.to_i
# end
#
# puts sum

# puts ARGF.reduce(0) { |sum, i| sum += i.to_i }
# puts ARGF.to_a.map(&:to_i).reduce(:+) # Inefficient because we iterate twice

def sum_file(filename)
 File.open(filename).reduce(0) { |sum, i| sum += i.to_i }
end

puts sum_file('numbers.txt')
